test/outpace/config_test.clj

```diff
@@ -55,6 +56,45 @@
       (is (nil? (extract ev)))
       (is (not (provided? ev))))))
 
+(deftest test-PropVal
+  (let [name     "name"
+        value    "value"
+        defined? true
+        ev       (->PropVal name value defined?)]
+    (testing "PropVal fields"
+      (is (= name (:name ev)))
+      (is (= value (:value ev)))
+      (is (= defined? (:defined? ev))))
+    (testing "PropVal protocol implementations"
+      (is (= value (extract ev)))
+      (is (= defined? (provided? ev))))
+    (testing "PropVal edn-printing"
+      (is (= (str "#config/property " (pr-str name)) (pr-str ev))))))
+
+(defn prop-var-name []
+  (let [name (first (keys (java.lang.System/getProperties)))]
+    (assert name "Cannot test read-property without environment variables")
```

Comment:
Should probably read "...without properties" though it may not even make sense to do this check? I think there are always properties defined?